hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/policies/FairSharePolicy.java

```diff
@@ -178,4 +179,15 @@ public boolean checkIfUsageOverFairShare(Resource usage, Resource fairShare) {
   public byte getApplicableDepth() {
     return SchedulingPolicy.DEPTH_ANY;
   }
+
+  @Override
+  public boolean isChildPolicyAllowed(SchedulingPolicy childPolicy)
+      throws AllocationConfigurationException {
+    if (childPolicy instanceof DominantResourceFairnessPolicy) {
```

Comment:
Based on my earlier comment, this should be `return ! (childPolicy instanceof DominantResourceFairnessPolicy`